src/service.ts

```diff
@@ -292,31 +287,27 @@ export class Service {
     return JSON.parse(await response.text());
   }
 
-  static saveJSON(json: object, uri: string): Promise<string> {
+  static async saveJSON(json: object, uri: string): Promise<string> {
     const update = !!uri;
-    return new Promise((resolve, reject) => {
-      const xhr = new XMLHttpRequest();
-      xhr.addEventListener("load", function() {
-        if (update) {
-          resolve(uri);
-        } else {
-          let jsonURI = JSON.parse(this.response).uri;
-          jsonURI = jsonURI.substring(jsonURI.lastIndexOf("/") + 1);
-          resolve(jsonURI);
-        }
+    if (update) {
+      const url = "//api.myjson.com/bins/" + uri;
+      const response = await fetch(url, {
+        method: "PUT",
+        headers: new Headers({ "Content-type": "application/json; charset=utf-8" }),
+        body: JSON.stringify(json)
       });
-      xhr.addEventListener("error", function() {
-        reject();
+      const result = JSON.parse(await response.text());
+      return uri;
+    } else {
+      const url = "//api.myjson.com/bins/";
+      const response = await fetch(url, {
+        method: "POST",
+        headers: new Headers({ "Content-type": "application/json; charset=utf-8" }),
+        body: JSON.stringify(json)
       });
-      if (update) {
-        xhr.open("PUT", "//api.myjson.com/bins/" + uri, true);
-      } else {
-        xhr.open("POST", "//api.myjson.com/bins", true);
-      }
-      xhr.setRequestHeader("Content-type", "application/json; charset=utf-8");
-      xhr.send(JSON.stringify(json));
-    });
+      return JSON.parse(await response.text()).uri;
   }
+}
```

Comment:
Two last '}' is miss aligned.

Reply:
I think its good because there are no linting errors and it looks good on my side. I don't know why the gist is not redirected as in the console it says `Gist Created`. May be the problem is with this part:`jsonURI = jsonURI.substring(jsonURI.lastIndexOf("/") + 1);`

Reply:
I think its good because there are no linting errors and it looks good on my side. I don't know why the gist is not redirected as in the console it says Gist Created. May be the problem is with this part:jsonURI = jsonURI.substring(jsonURI.lastIndexOf("/") + 1);

Reply:
> I think its good because there are no linting errors and it looks good on my side. 

No errors are catch-able by linter, e.g. https://github.com/wasdk/WebAssemblyStudio/pull/125/files#r169961156. You need to use `git show` command to (re)inspect what will be submitted to the reviewer. That's where regular commit squashing is useful.